src/org/folio/rest_v2/Eureka.groovy

```diff
@@ -0,0 +1,60 @@
+package org.folio.rest_v2
+
+import org.folio.models.OkapiTenant
+
+class Eureka extends Authorization {
+
+  public OkapiTenant superTenant
+
+  Eureka(Object context, String okapiDomain, OkapiTenant superTenant, boolean debug = false) {
+    super(context, okapiDomain, debug)
+    this.superTenant = superTenant
+  }
+
+
+  void registerApplication(List descriptorsList, OkapiTenant tenant, String applicationId) {
+
+    String checkUrl = generateUrl("/applications/${applicationId}")
+    def response = restClient.get(checkUrl, headers).body
+    int statusCode = response.status
+
+    if (statusCode == 200) {
```

Comment:
better to move this check into separate method like Okapi.isTenantExist() in Okapi.createTenant() method

Reply:
Fixed